### DISCUSSION TITLE
No keyboard input ... Module Search box inactive

### DISCUSSION THREAD

**Discussion by NoiseExplorer** [DISTRHO/Cardinal#261] [Q&A]
Hi - Firstly, thanks a million to all involved in offering Cardinal to the world as a huge, free playground!  Secondly, I'm a newbie so apologies if this is a numpty question, but ... I'm unable to use the Search box in the Module browser as the field is greyed out.  Furthermore, my computer keyboard has no input within my Cardinal session - no keystrokes are recognised, (e.g. keyboard shortcuts to delete, bypass, duplicate modules etc.), even after selecting option to 'give all keystrokes to plugin.  I'm using the Synth variant as a VST3 (vers 22.05) running inside Cakewalk (Windows 10 x64 bit).  Apologies if this is not the right forum to post, but any help would be much appreciated.  Thanks

**Comment by dromer** [comment]
The `Discussions` tab is meant for proposing features and additions to the project.

If you have an Issue please first search in the `Issues` tab for any related problems. So that we don't duplicate and clutter the backlog.

Likely your issue relates to one of these tickets: https://github.com/DISTRHO/Cardinal/issues?q=is%3Aissue+is%3Aopen+keyboard
All VST3 related problems need to be focused on this ticket: https://github.com/DISTRHO/Cardinal/issues/41
The VST3 is still beta and should not be used for production!

Recommend to close this discussion.

**Comment by falkTX** [comment]
We can have as many discussions as needed, they will clutter things a bit, but personally I dont care.

That said, retelling reminder that VST3 in Cardinal is still experimental.
Prefer to use LV2, VST2 or AU.

**Reply by dromer** [reply]
There is nothing here to be discussed, it was a misunderstanding of the troubleshooting process. Both issue topics belong elsewhere I think.

**Comment by NoiseExplorer** [comment]
@dromer and @falkTX - thank you both for responding.  Apologies as I'm new to Github and hadn't clocked the 'issues' page.  Comments re VST3 in beta are noted and I'll take a look at the ticket referenced by dromer.  Thanks again

**Comment by subderground** [comment]
When Cardinal is in the active window: 
-VST2 version, all the keyboard is still active. So you can type correctly the words in "search" and in a "label" module. But you can't hit the "spacebar" to play/stop the DAW (To do that you to have to click outside to unactive the Cardinal window).
-VST3 version,  you can play/stop the DAW with the spacebar while Cardinal is in the active window, but these kind of keys (spacebar-del-backspace) are still unusable in "search" and in a "label" module.

So I think the spacebar should  be assigned to play/stop the DAW (like classic vst plugins). But when clicking in a text area ("search" or into a module texte area), the spacebar-del-backspace should switch to be assigned to the text typing. My 2 cents :-)

**EDIT: Everything is now fixed with the 22.07  VST & VST3 versions. Great work !!!**

**Reply by dromer** [reply]
Interesting idea, the latter would certainly be a nice UX. being able to control daw transport while having focus on cardinal would be nice. Input boxes (and transport control) should ideally at least work the same for all formats, platforms and hosts. It's not always clear how/why certain odd behavior/implementation block certain custom features, though.

**Comment by mwodring** [comment]
> Hi - Firstly, thanks a million to all involved in offering Cardinal to the world as a huge, free playground! Secondly, I'm a newbie so apologies if this is a numpty question, but ... I'm unable to use the Search box in the Module browser as the field is greyed out. Furthermore, my computer keyboard has no input within my Cardinal session - no keystrokes are recognised, (e.g. keyboard shortcuts to delete, bypass, duplicate modules etc.), even after selecting option to 'give all keystrokes to plugin. I'm using the Synth variant as a VST3 (vers 22.05) running inside Cakewalk (Windows 10 x64 bit). Apologies if this is not the right forum to post, but any help would be much appreciated. Thanks

Right click - select all. RIght click = delete. Takes an extra second but will solve your problem easy. : )

**Reply by NoiseExplorer** [reply]
Thanks mwodring ... The bigger challenge is searching for items in the module library 😉. But as has been fairly highlighted, I'm using the beta version so I'll just remain patient for now, and grateful that there are others out there who are far more talented than me and who are generous enough to offer this for free 😊

**Comment by subderground** [comment]
For me, everything is fixed now with 22.07 version (vst vst3).
Great work ! Thanks.
